modules/gmc-agent/src/main/java/org/gridgain/action/controller/SecurityActionsController.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gridgain.action.controller;
+
+import org.apache.ignite.IgniteAuthenticationException;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.plugin.security.SecurityCredentials;
+import org.gridgain.action.annotation.ActionController;
+import org.gridgain.agent.AgentUtils;
+import org.gridgain.dto.action.AuthenticateCredentials;
+import org.gridgain.action.Session;
+import org.gridgain.action.SessionRegistry;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Controller for security actions.
+ */
+@ActionController("SecurityActions")
+public class SecurityActionsController {
+    /** Context. */
+    private final GridKernalContext ctx;
+
+    /** Registry. */
+    private final SessionRegistry registry;
+
+    /** Logger. */
+    private final IgniteLogger log;
+
+    /**
+     * @param ctx Context.
+     */
+    public SecurityActionsController(GridKernalContext ctx) {
+        this.ctx = ctx;
+        log = ctx.log(SecurityActionsController.class);
+        registry = SessionRegistry.getInstance(ctx);
+    }
+
+    /**
+     * @param reqCreds Request credentials.
+     */
+    public CompletableFuture<String> authenticate(AuthenticateCredentials reqCreds) {
+        CompletableFuture<String> fut = new CompletableFuture<>();
+
+        try {
+            Session ses = authenticate0(reqCreds);
+            registry.saveSession(ses);
+
+            if (log.isDebugEnabled())
+                log.debug("Next sessionId were generated for request: " +
+                        "[sessionId=" + ses.id() + "]");
```

Comment:
Lets use message: "Session ID was generated for request: [sessionId=" + ses.id() +", req=" + req + req.getId() + "]"

Reply:
Fix, but we don't have request object here.